src/shims/x86/mod.rs

```diff
@@ -408,3 +615,84 @@ fn horizontal_bin_op<'tcx>(
 
     Ok(())
 }
+
+/// Conditionally multiplies the packed floating-point elements in
+/// `left` and `right` using the high 4 bits in `imm`, sums the four
+/// products, and conditionally stores the sum in `dest` using the low
```

Comment:
Since this is "conditional", I assume these are not actually four products but up to four products?